areader/reader.go

```diff
@@ -861,6 +862,18 @@ func (ar *Reader) readAndInstall(r io.Reader, i handlers.Installer,
 			return errors.Errorf("Payload: can not find data file: %s", hdr.Name)
 		}
 
+		matched, err := regexp.MatchString(`^[\w\-.,]+$`, filepath.Base(hdr.Name))
+
+		if err != nil {
+			return errors.Wrapf(err, "Payload: invalid regular expression pattern")
+		}
+
+		if !matched {
+			message := "Payload: data file " + hdr.Name + " contains forbidden characters"
+			info := "Only letters, digits and characters in the set \".,_-\" are allowed"
+			return fmt.Errorf(message + "\n" + info)
```

Comment:
Newlines don't work well with our logging library, they will show up as literal `\n`. Better to just use period and space to separate.

Also, this will not behave correctly if `hdr.Name` contains for example `%s`. The more idiomatic way would be:

```
fmt.Errorf("%s %s", message, info)
```

This assumes that `message` ends with a period.

Reply:
@kacf Done in both files. The idiomatic syntax applied.